### ISSUE TITLE
Add-ons with localized names are not correctly listed among search/?sort=name results

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons-server#7600]
STR:

1. Send a search request that will return extensions, sorted by name
```
GET https://addons-dev.allizom.org/api/v3/addons/search/?sort=name&type=extension&tag=firefox57&page=8
```
2. Look at the search results and locate the following add-ons:  **id 492376** and **id 494543** 
3. Check the names for these two add-ons and compare them with the other names listed in the results

Actual result:
The two add-ons (english) names are starting with letter "G" respectively "D" while all other results are starting with "C"

Expected results:
The two mentioned add-ons should not be listed when performing the above search request

Notes:
- both add-ons had an English (en-US) translation in their locales file and both had default_locale set to en-US.
- per @diox: I think it's because we don't explicitly set which translation to use when indexing the name we use for sorting purposes (for the affected add-ons, some translations do start with a "C") .
- reproduced with AMO-dev, FF58, Win10x64

**Comment by diox** [comment]
I think we should be forcing activation of either `settings.LANGUAGE_CODE` locale or whatever the add-on default locale is when we index add-ons.

**Comment by EnTeQuAk** [comment]
> I think we should be forcing activation of either settings.LANGUAGE_CODE locale or whatever the add-on default locale is when we index add-ons.

Agreed but that'll still result in weird results during sorting, e.g if we don't have an entry for that add-on in the requested language by the user right? I'm not sure if we can avoid that, it could very well be a missing "locale not available in your language" warning or so to make it more visible?

**Comment by diox** [comment]
> Agreed but that'll still result in weird results during sorting, e.g if we don't have an entry for that add-on in the requested language by the user right?

Well, it could be weird, but at least it'll be more predictable than it is right now.

> I'm not sure if we can avoid that, it could very well be a missing "locale not available in your language" warning or so to make it more visible?

I don't think we should present any warning or message to the user about it. It's a niche feature (we don't even expose it on the frontend currently), and as long as the order is stable, I think it's ok.

**Comment by diox** [comment]
We now store a raw variant of all translations of the add-on name (it was needed for https://github.com/mozilla/addons-server/issues/6898) so we /could/ follow translations for sorting if we wanted to, now.

**Comment by AlexandraMoga** [comment]
I've tried to reproduce what was mentioned above (note that the request is no longer valid since many things have changed in between) but after a quick scan I haven't noticed this problem anymore.

Specifically, the add-on IDs that were causing problems in the above report are now listed where they are expected to be when the `sort=name` param is present. 

I'm not sure if the issue was fixed by chance since reported, or the above scenario no longer catches it. 

I'm going to close this as works for me and will re-open if I manage to reproduce it again.
